crates/rustc_codegen_spirv/src/builder_spirv.rs

```diff
@@ -425,6 +421,18 @@ impl BuilderSpirv {
             })
     }
 
+    pub fn has_extension(&self, extension: &str) -> bool {
+        self.builder
+            .borrow()
+            .module_ref()
+            .extensions
+            .iter()
+            .any(|inst| {
+                inst.class.opcode == Op::Extension
+                    && inst.operands[0].unwrap_literal_string() == extension
+            })
```

Comment:
This can be `O(1)` as a set of interned strings, maybe open an issue about simple algorithmic inefficiencies like this and leave `// FIXME(#1234)` comments? At least that's what I ~~would~~ should do.